users/models.py

```diff
@@ -153,6 +157,13 @@ def is_employee(self) -> bool:
     def get_reports_created(self) -> QuerySet:
         return self.report_set.select_related("task_activities").order_by("-date", "project__name")
 
+    def get_project_ordered_by_last_report_creation_date(self) -> QuerySet:
+        return self.projects.annotate(
+            last_report_creation_date=Coalesce(
+                Max("report__creation_date", filter=Q(report__author=self)), Value("1970-01-01 00:00:00")
```

Comment:
I am not sure if I understand why we need `Coalesce` and `Max` here... Could you please explain?

Reply:
If given project has no reports created for current user, the returned value from `Max()` is `None, which then is being put at the beginning of the list (because we use DESC ordering). So to avoid that I am using `Coalesce` to replace `None` with oldest date known to human civilization and get the desired order.